api/src/main/java/org/apache/iceberg/Table.java

```diff
@@ -44,6 +44,15 @@
    */
   TableScan newScan();
 
+
+  /**
+   * @param fromSnapshotId - the last snapshot id read by the user, exclusive
+   * @param toSnapshotId - read incremental data upto this snapshot id
+   * @return a table scan which can read incremental data from {@param fromSnapshotId}
+   * exclusive and up to {@toSnapshotId} inclusive
+   */
+  TableScan newIncrementalScan(long fromSnapshotId, long toSnapshotId);
```

Comment:
You may also consider adding `newIncrementalScan(long fromSnapshotId)` that uses `currentSnapshot` for `toSnapshotId`. We would need a method to get the scan's "to" snapshot, though, so that it can be saved as reader state.

Reply:
I'm not sure whether {{newDeltaScan}} fits here. Since this is not a pure delta and is more closely resembles only the appends between two snapshots [E.g as commented below maybe we will have to throw exceptions for overwrites and deletes?]

Maybe it is best to rename it as {{appendsBetween}} as @rdblue mentions?

Reply:
Agreed. `appendsBetween` works, and since it is clear from the name that only appends are to be returned, I don't think you'd need to throw if you find overwrites or deletes, you could ignore.

I think we would need `appendsAfter(long fromSnapshotId)` as well, for  the case that would read till current snapshot?


Side note: `deletesBetween` and `overwritesBetween` APIs would be nice as well in the future. We could help with that.

Reply:
Fixed. Added a newer api, and renamed the methods to `appendsBetween` and `appendsAfter`